tracing-attributes/src/attr.rs

```diff
@@ -268,6 +268,48 @@ impl Default for FormatMode {
     }
 }
 
+impl FormatMode {
+    /// Returns a list of the valid options for the format mode, this should be updated
+    /// when new modes are added to ensure that parser error messages are formatted correctly.
+    fn options() -> &'static [&'static str] {
+        &["Debug", "Display", "Value"]
+    }
+}
+
+impl<S> From<S> for FormatMode
+where
+    S: AsRef<str>,
+{
+    fn from(s: S) -> Self {
+        match s.as_ref() {
+            "Debug" => FormatMode::Debug,
+            "Display" => FormatMode::Display,
+            "Value" => FormatMode::Value,
+            _ => FormatMode::default(),
+        }
+    }
+}
```

Comment:
I think the refactor to make it easier to add new variants makes sense, but the new variant still needs to be added in two places, once in the list of `options` and once in `From<S>`. What do you think about having the list of options being a `&[(&'static str, FormatMode)]`, and changing the `From<S>` impl to look like this:
```suggestion
impl FormatMode {
    /// A list of the valid options for the format mode. This should be updated
    /// when new modes are added to ensure that parser error messages are formatted correctly.
    const OPTIONS: &[(&str, FormatMode)] = &[
        ("Debug", FormatMode::Debug),
        ("Display", FormatMode::Display),
        ("Value", FormatMode::Value),
    ];
}

impl<S> From<S> for FormatMode
where
    S: AsRef<str>,
{
    fn from(s: S) -> Self {
        for (name, value) in Self::OPTIONS {
            if s == name {
                 return value;
            }
        }
        
        FormatMode::default()
    }
}
```

This way, new variants need _only_ be added to the list of options, and the `FromStr` implementation doesn't need to be changed every time. 

Not a blocker, if this complicates the implementation too much, I'm fine with the current approach --- it's just a thought.

Reply:
Happy enough to adopt this approach :+1: